### ISSUE TITLE
ci-operator failures when using 4.3-art-latest in tag_specification

### ISSUE THREAD

**Issue by petr-muller** [openshift/ci-tools#317]
TL;DR: ci-operator fails to operate when `4.3-art-latest` is used in `tag_specification` stanza. This is because it iterates over `.Spec.Tags` of that imagestream and `Spec` is empty/changing in `4.3-art-latest`. It looks like ci-operator should iterate over `.Status.Tags` but it is not clear to me whether the current behavior is a bug or a feature, and whether such change would be a risk.

---

Gory details:

This PR https://github.com/openshift/release/pull/6254 was attempting to add a ci-operator config using `4.3-art-latest` imagestream in its `tag_specification` stanza (advised by Steve):

```
tag_specification:
  name: "4.3-art-latest"
  namespace: ocp
```

Rehearsals of test jobs generated from such configs [failed](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/6254/rehearse-6254-pull-ci-openshift-cnv-cnv-ci-master-e2e-test/7) with the following error (see the full error in [1]):
```
parameter RELEASE_IMAGE_LATEST is required and must be specified
```
This was [reported on Slack](https://coreos.slack.com/archives/CBN38N3MW/p1575894715236400). We discovered that ci-operator is not building a release payload like usual, accompanied by the following [log output](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/6254/rehearse-6254-pull-ci-openshift-cnv-cnv-ci-master-e2e-test/7#1:build-log.txt%3A15):

```
No latest release image necessary, stable image stream does not include a cluster-version-operator image
```

(This is likely a first issue to fix: if we need a release, we should either have `RELEASE_IMAGE_*` set, or error out on `...stable image stream does not include...` instead of continuing)

Manual inspection in ci-operator namespace showed that their `stable` imagestreams contained no images; we expected images from `4.3-art-latest` to be tagged there. This varied a lot: in other runs, we saw e.g. only `machine-os-content` image tagged in `stable`, and once, we even saw all expected images tagged there and `ci-operator` even attempted to assemble a release payload.

It looks like ci-operator iterates over `.Spec.Tags` of the source imagestream, followed by some filtering/validation based on the image's presence in its `.Status`:

https://github.com/openshift/ci-tools/blob/766aafaf8fbdec94ed85958ee700a36c02589961/pkg/steps/release/release_images.go#L183-L197

We asked [ART on Slack, Justin Pierce suggested](https://coreos.slack.com/archives/CB95J6R4N/p1575902450475900) ci-operator should use `Status.Tags` instead of `.Spec.Tags`, but given that ci-operator now basically uses both, it's not clear to me if the change is really that simple and what risk it implies.

/cc @stevekuznetsov @lioramilbaum @droslean 

---
[1] linebreaks mine
```
could not wait for template instance to be ready: 
  could not determine if template instance was ready: 
  failed to create objects: 
  Template.template.openshift.io "e2e-test" is invalid: 
  template.parameters[10]: Required value: 
  template.parameters[10]: 
  parameter RELEASE_IMAGE_LATEST is required and must be specified
```

**Comment by smarterclayton** [comment]
CI operator uses spec tags and status tags for different things.  It should work with only status tags (status tag is authoritative), as several parts already do

**Comment by smarterclayton** [comment]
Note in some places spec tags provide additional info / context, and you need to understand those cases.
